core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -2084,7 +2087,8 @@ public boolean processWalBlock(
                         // Can commit without O3 and LAG has just enough rows
                         || (commitToTimestamp >= newMaxLagTimestamp && totalUncommitted > getMetaMaxUncommittedRows())
                         // Too many uncommitted transactions in LAG
-                        || (configuration.getWalMaxLagTxnCount() > 0 && txWriter.getLagTxnCount() >= configuration.getWalMaxLagTxnCount());
+                        || (configuration.getWalMaxLagTxnCount() > 0 && txWriter.getLagTxnCount() >= configuration.getWalMaxLagTxnCount())
+                        || (configuration.getMicrosecondClock().getTicks() - lastWalCommitTimestampMicros > configuration.getCommitLatency());
```

Comment:
nit: would be nice to put a clarifying comment similar to the above checks.

Reply:
done in https://github.com/questdb/questdb/pull/5118/commits/162c533b42c0cb80ae9418ca616e92b57ea8d839